File: fwechelon-api/src/main/kotlin/it/forgottenworld/echelonapi/FWEchelon.java
package it.forgottenworld.echelonapi;

import org.bukkit.Bukkit;
import org.bukkit.plugin.Plugin;
import org.jetbrains.annotations.Nullable;

public interface FWEchelon {

    /**
     * Convenience method for retrieving the FWEchelonPlugin's API
     *
     * @return the plugin's API or null if the plugin is not loaded
     */
    @Nullable
    static FWEchelonApi getApi() {
        Plugin plugin = Bukkit.getPluginManager().getPlugin("FWEchelon");
        return plugin instanceof FWEchelonApi ? (FWEchelonApi) plugin : null;
    }

}

File: fwechelon-api/src/main/kotlin/it/forgottenworld/echelonapi/FWEchelon.kt
package it.forgottenworld.echelonapi

import org.bukkit.Bukkit

object FWEchelon {

    /**
     * Convenience method for retrieving FWEchelonPlugin's API
     *
     * @return the plugin's API
     * @throws [IllegalStateException] if the plugin is not loaded
     */
    @JvmStatic
    val api get() = Bukkit
        .getPluginManager()
        .getPlugin("FWEchelon") as? FWEchelonApi
        ?: error("FWEchelon not loaded")
}